### PR TITLE
[docs] Add documentation for .easignore file

### PR DESCRIPTION
**Pull request by betomoedano**
# Why

This solves [ENG-9211](https://linear.app/expo/issue/ENG-9211/add-documentation-for-easignore).

# How

Create a small page explaining why and how to use a `.easignore` file.

# Test Plan

Ran docs locally and validate everything looks good. 

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build-reference/easignore.mdx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+<Step label="2">
+
+Copy the content of the `.gitignore` file into the `.easignore` file. Then, add any files that are unnecessary for the build process.
+
+```bash .easignore
+# Copy everything from your .gitignore file here
+
+# Additional ignores specific to EAS builds
+
+# Ignore native folders (If you are using CNG)
```
Comment (amandeepmittal) [inline docs/pages/build-reference/easignore.mdx:32]:
```suggestion
# Ignore native folders (if you are using CNG)
```

File: docs/pages/build-reference/easignore.mdx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/ios
+/android
```
Comment (amandeepmittal) [inline docs/pages/build-reference/easignore.mdx:34]:
```suggestion
/android
/ios
```

Let's also use the [correct platform order](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#platform-order-when-referencing-android-ios-and-web-or-multiple-platforms).

File: docs/pages/build-reference/easignore.mdx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+<Step label="3">
+
+Save the file and trigger a new build.
+
+<Terminal cmd={['$ eas build --platform ios --profile development']} />
+
+</Step>
+
+Congrats! You've successfully configured your `.easignore` file.
```
Comment (amandeepmittal) [inline docs/pages/build-reference/easignore.mdx:52]:
```suggestion
You've successfully configured your **.easignore** file.
```

File: docs/pages/build-reference/easignore.mdx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+# Ignore native folders (If you are using CNG)
+/ios
+/android
+
+# Ignore test coverage reports
+/coverage
+```
+
+Learn more about [Continuous Native Generation (CNG)](https://docs.expo.dev/workflow/continuous-native-generation).
```
Comment (amandeepmittal) [inline docs/pages/build-reference/easignore.mdx:40]:
```suggestion
Learn more about [Continuous Native Generation (CNG)](/workflow/continuous-native-generation).
```

I think we should add some context on why someone should learn at CNG. We can briefly explain here the comment from the above snippet on why to add android and ios directories to this file and the link to the CNG guide in the context.

**Comment by betomoedano** [comment]
I changed CNG to EAS Build; it is more straightforward to say that if users don't include Android and iOS folders, EAS will run Prebuild to generate the native directories.

File: docs/pages/build-reference/easignore.mdx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+/android
+/ios
+
+# Ignore test coverage reports
+/coverage
+```
+
+If your project does not contain android and ios directories, EAS Build will run Prebuild to generate these native directories before compilation.
+
+Learn more about [EAS Build](/workflow/prebuild/#usage-with-eas-build).
```
Comment (amandeepmittal) [inline docs/pages/build-reference/easignore.mdx:40]:
```suggestion
```

This line and the link seem a bit out of context. Let's use this link in the above line.

File: docs/pages/build-reference/easignore.mdx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+# Ignore native folders (if you are using EAS Build)
+/android
+/ios
+
+# Ignore test coverage reports
+/coverage
+```
+
+If your project does not contain android and ios directories, EAS Build will run Prebuild to generate these native directories before compilation.
```
Comment (amandeepmittal) [inline docs/pages/build-reference/easignore.mdx:38]:
```suggestion
If your project does not contain **android** and **ios** directories, [EAS Build will run Prebuild](/workflow/prebuild/#usage-with-eas-build) to generate these native directories before compilation.
```

**Review by amandeepmittal — APPROVED** [review]
Lgtm 👍 

There's a formatting error due to the last change. Please resolve that.

**Review by Simek — APPROVED** [review]
LGTM, remember to fix lint before shipping!

**Comment by betomoedano** [comment]
Thanks for reviewing 🙏

**Pull request merged**